extensions/amp-gwd-animation/0.1/amp-gwd-animation-impl.js

```diff
@@ -0,0 +1,447 @@
+   * @param {!Element} receiver
+   * @param {string} label
+   * @private
+   */
+  playLabelAnimation_(receiver, label) {
+    // Unpause playback.
+    receiver.classList.remove(PlaybackCssClass.PAUSE);
+
+    // If another goto animation is currently active on this element, stop it.
+    const currentLabel =
+        receiver.getAttribute(CURRENT_LABEL_ANIMATION_ATTR);
+
+    if (currentLabel) {
+      receiver.classList.remove(currentLabel);
+      receiver.removeAttribute(CURRENT_LABEL_ANIMATION_ATTR);
+    }
+
+    // A forced reflow is needed if removing and readding the same class so
+    // the animation is restarted.
+    if (currentLabel == label) {
+      reflow(receiver);
+    }
+
+    // Add the label animation class and record this label as this element's
+    // current animation.
+    receiver.classList.add(label);
+    receiver.setAttribute(CURRENT_LABEL_ANIMATION_ATTR, label);
+  }
+
+  /**
+   * Handles GWD event `animationend` events, which signal that a timeline event
+   * marker has been reached.
+   * If the event originated from a GWD event element, extracts its event name
+   * and dispatches a custom event.
+   * @param {!Event} event An `animationend` event.
+   * @private
+   */
+  onAnimationEndEvent_(event) {
+    const userEventName = event.target.getAttribute(EVENT_NAME_ATTR);
+
+    if (!userEventName) {
+      // No GWD event name could be extracted, usually just because the event is
+      // from some other animation. Just ignore it.
+      return;
+    }
+
+    const timelineEvent = createCustomEvent(
+        this.ampdoc_.win,
+        `${GWD_TIMELINE_EVENT}`,
+        {eventName: userEventName, sourceEvent: event});
+
+    this.ampdoc_.win.dispatchEvent(timelineEvent);
+  }
+
+  /**
+   * @private
+   */
+  listenForAnimationEnd_() {
+    for (let i = 0; i < VENDOR_ANIMATIONEND_EVENTS.length; i++) {
+      this.ampdoc_.getBody().addEventListener(
+          VENDOR_ANIMATIONEND_EVENTS[i], this.boundOnAnimationEndEvent_, true);
+    }
+  }
+
+  /**
+   * @private
+   */
+  unlistenForAnimationEnd_() {
+    for (let i = 0; i < VENDOR_ANIMATIONEND_EVENTS.length; i++) {
+      this.ampdoc_.getBody().removeEventListener(
+          VENDOR_ANIMATIONEND_EVENTS[i], this.boundOnAnimationEndEvent_, true);
+    }
+  }
+
+  /** @override */
+  dispose() {
+    this.unlistenForAnimationEnd_();
+  }
+}
+
+/**
+ * Utility function which triggers layout on the element. This is necessary
+ * to invoke when reapplying the same label animation class so the animation
+ * restarts.
+ * @param {!Element} element
+ */
+const reflow = function(element) {
+  element./*OK*/offsetWidth = element./*OK*/offsetWidth;
+};
+
+/**
+ * Registers and immediately starts the GWD runtime service for the given
+ * ampdoc. Retrieve the service instance with getServiceForDoc and
+ * GWD_SERVICE_NAME.
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+ */
+export const installGwdRuntimeServiceForDoc = function(ampdoc) {
```

Comment:
This is optional for now, but may be needed if you plan for this service to be used w/o the `amp-gwd-animation` element.

This changed a bit, and now looks like this: https://github.com/ampproject/amphtml/blob/db824d0f7f42d75444df1e97461d789819481faf/extensions/amp-dynamic-css-classes/0.1/amp-dynamic-css-classes.js#L155

Could you please change? That usually means that you may also need to include extension explicitly in test like this: https://github.com/ampproject/amphtml/blob/master/extensions/amp-3q-player/0.1/test/test-amp-3q-player.js#L25

On the plus side this provides for better test isolation.

Reply:
Updated to use registerServiceForDoc().